sources/engine/Stride.Input/Windows/RawInput/RawInput.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Stride contributors (https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Stride.Input.RawInput
+{
+    internal static class RawInput
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        private struct RawInputDevice
+        {
+            public ushort UsagePage;
+
+            public ushort Usage;
+
+            public uint Flags;
+
+            public IntPtr Target;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RawInputHeader
+        {
+            public int dwType;
+            public int dwSize;
+            public IntPtr hDevice;
+            public IntPtr wParam;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RawMouse
+        {
+            [StructLayout(LayoutKind.Explicit)]
+            public struct RawMouseButtonsData
+            {
+                [FieldOffset(0)]
+                public int Buttons;
+
+                [FieldOffset(0)]
+                public short ButtonFlags;
+
+                [FieldOffset(2)]
+                public short ButtonData;
+            }
+
+            public ushort Flags;
+            public RawMouseButtonsData ButtonsData;
+            public int RawButtons;
+            public int LastX;
+            public int LastY;
+            public int ExtraInformation;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RawKeyboard
+        {
+            public ushort MakeCode;
+            public ushort Flags;
+            public ushort Reserved;
+            public ushort VKey;
+            public uint Message;
+            public uint ExtraInformation;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RawHid
+        {
+            public uint dwSizeHid;
+            public uint dwCount;
+            public int bRawData;
+        }
+
+        [StructLayout(LayoutKind.Explicit)]
+        public struct RawInputDataInner
+        {
+            [FieldOffset(0)]
+            public RawMouse Mouse;
+
+            [FieldOffset(0)]
+            public RawKeyboard Keyboard;
+
+            [FieldOffset(0)]
+            public RawHid Hid;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct RawInputData
+        {
+            public RawInputHeader header;
+            public RawInputDataInner data;
+        }
+
+        public static unsafe byte[] GetHIDRawData(ref RawHid rawHid)
+        {
+            var byteCount = rawHid.dwCount * rawHid.dwSizeHid;
+            var result = new byte[byteCount];
+            fixed (byte* dest = result)
+            {
+                fixed (int* src = &rawHid.bRawData)
+                {
+                    Unsafe.CopyBlockUnaligned(dest, src, byteCount);
```

Comment:
Hmm, that is a weird one. It is most definitely available under netfw. I tested it both under netcore and netfw and it seems to work fine on both. Don't know why the docs say that it isn't available on netfw.

Reply:
Built a simple project under ``<TargetFramework>net472</TargetFramework>`` containing that signature
and it returns ``error CS0103: The name 'Unsafe' does not exist in the current context``

Reply:
Huh, weird. Pretty sure it compiled here yesterday, but I might be mistaken. I will check again tonight. Worst case scenario that part of the code can be removed completely because it handles HID input which isn't used. I only added it to completely handle the WM_INPUT message.

Reply:
After a bit of research I think I found the cause. `System.Runtime.CompilerServices.Unsafe` is being included as a transient package by another dependency(don't know which one yet) so that is why that class is available under netfw. I will add it as a direct dependency.